TheaterSchedule/Controllers/PostersController.cs

```diff
@@ -13,17 +16,28 @@ namespace TheaterSchedule.Controllers
     public class PostersController : ControllerBase
     {
         IPostersService postersService;
-        public PostersController(IPostersService postersService)
+        IMemoryCache memoryCache;
+
+        public PostersController(IPostersService postersService, IMemoryCache memoryCache)
         {
             this.postersService = postersService;
+            this.memoryCache = memoryCache;
         }
 
         // GET: api/Posters
         [HttpGet("{languageCode}")]
-        public JsonResult Get(string languageCode)
+        public async Task<List<PostersDTO>> Get(string languageCode)
         {
-            return new JsonResult(postersService.LoadPostersData(languageCode));
+            List<PostersDTO> performancesData = null;
+            string memoryCacheKey = "PerformanceList";

```

Comment:
why key has no language code?
if first request will be 'ua' then in cache will be stored ua translation
For next request with 'en' will be returned cached 'ua' performances